### PR TITLE
[12.0][IMP] stock_barcodes: Allow get lots automatically depends on removal strategy for picking operations

### PR DESCRIPTION
**Pull request by sergio-teruel**
cc @Tecnativa  TT31666

ping @carlosdauden @chienandalu

### CODE REVIEW

File: stock_barcodes/wizard/stock_barcodes_read.py [hunk 1]
```diff
@@ -56,6 +59,12 @@ class WizStockBarcodesRead(models.AbstractModel):
         ('success', 'Barcode read correctly'),
     ], readonly=True)
     message = fields.Char(readonly=True)
+    auto_lot = fields.Boolean(
+        string="Get lots automatically",
+        help="If checked the lot will be set automatically with the same "
+             "removal startegy",
+        default=lambda self: self._default_auto_lot(),
```
Comment (carlosdauden) [inline stock_barcodes/wizard/stock_barcodes_read.py]:
```suggestion
        default=_default_auto_lot,
```

**Review by carlosdauden — APPROVED** [review]

**Comment by sergio-teruel** [comment]
Changes done!!!

File: stock_barcodes/models/res_config_settings.py [hunk 2]
```diff
@@ -22,4 +22,4 @@ class ResConfigSettings(models.TransientModel):
 class ResCompany(models.Model):
     _inherit = "res.company"
 
-    stock_barcodes_inventory_auto_lot = fields.Boolean()
+    stock_barcodes_auto_lot = fields.Boolean()
```
Comment (chienandalu) [inline stock_barcodes/models/res_config_settings.py:25]: A migration script should be done, isn't it?

**Review by chienandalu — COMMENTED** [review]
I couldn't do the auto lot in a delivery picking (inventory works as expected):

Selección_01051

File: stock_barcodes/models/res_config_settings.py (hunk 2 above)
Comment (sergio-teruel) [reply stock_barcodes/models/res_config_settings.py:25]: I think that is not mandatory... If user see the check to False only has to set to true in settings....

**Review by chienandalu — APPROVED** [review]
Working on production for serveral months without further issue :+1:

**Comment by pedrobaeza** [comment]
/ocabot merge minor

@chienandalu can you fw-port it to v13?

**Comment by chienandalu** [comment]
> @chienandalu can you fw-port it to v13?

Sure

**Pull request merged**